br/pkg/resolver/resolver.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolver
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/pingcap/tidb/br/pkg/restore"
+	grpcresolver "google.golang.org/grpc/resolver"
+)
+
+const (
+	schemePrefix           = "tikvstore-"
+	defaultGetStoreTimeout = time.Second * 30
+)
+
+type Builder struct {
+	scheme      string
+	splitClient restore.SplitClient
+}
+
+func NewBuilder(splitClient restore.SplitClient) *Builder {
+	return &Builder{
+		scheme:      generateScheme(),
+		splitClient: splitClient,
+	}
+}
+
+func generateScheme() string {
+	var buf [8]byte
+	_, err := io.ReadFull(rand.Reader, buf[:])
+	if err != nil {
+		panic(err)
+	}
+	return schemePrefix + hex.EncodeToString(buf[:])
```

Comment:
NIT: what about use pd-addr to generate this random suffix so different connecting to the cluster can also share the same resolver. randomize is not stable..in theory.

Reply:
Each resolver need to be bound to a PD client. If we want to share the same resolver, we need to share the same PD client as well. So a global PD client must be introduced. It looks a little weird.